File: app/src/main/java/com/example/androidcodelabapp/view/DetailActivity.java
package com.example.androidcodelabapp.view;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.androidcodelabapp.R;
import com.example.androidcodelabapp.model.GithubUsers;
import com.example.androidcodelabapp.presenter.GithubPresenter;

public class DetailActivity extends AppCompatActivity implements SingleDeveloperView {
    private String githubUserName;
    Button share;
    GithubUsers sharedInfo;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        getIncomingIntent();
        GithubPresenter presenter = new GithubPresenter();
        presenter.getDeveloperProfile(githubUserName, this
        );
        share = findViewById(R.id.sharebutton);
        share.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent shareIntent = new Intent();
                shareIntent.setAction(Intent.ACTION_SEND);
                shareIntent.setType("text/plain");
                shareIntent.putExtra(Intent.EXTRA_TEXT, String.format("Checkout this awesome developer @%s, %s.", sharedInfo.getUserName(), sharedInfo.getProfile()));
                startActivity(Intent.createChooser(shareIntent, String.format("Checkout this awesome developer @%s, %s.:", sharedInfo.getUserName(), sharedInfo.getProfile())));
            }

        });
    }

    private void getIncomingIntent() {
        if (getIntent().hasExtra("gitUserName") && getIntent().hasExtra("profileImage")) {
            githubUserName = getIntent().getStringExtra("gitUserName");
            String profileImage = getIntent().getStringExtra("profileImage");
            setProfile(githubUserName, profileImage);
        }
    }

    private void setProfile(String userName, String profileImage) {
        TextView name = findViewById(R.id.gitusername);
        name.setText(userName);
        ImageView image = findViewById(R.id.image);
        Glide.with(this).asBitmap().load(profileImage).into(image);


    }

    @Override
    public void showDeveloperProfile(GithubUsers profile) {
        sharedInfo = profile;
        View view = findViewById(R.id.detail_activity);
        TextView githubUrl = view.findViewById(R.id.githuburl);
        TextView organization = view.findViewById(R.id.org);

        githubUrl.setText(profile.getProfile());
        organization.setText(profile.getOrganization());


    }


}

File: app/src/main/java/com/example/androidcodelabapp/view/DetailActivity.kt
package com.example.androidcodelabapp.view

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.androidcodelabapp.databinding.ActivityDetailBinding
import com.example.androidcodelabapp.model.GithubUsers
import com.example.androidcodelabapp.presenter.GithubPresenter

class DetailActivity : AppCompatActivity(), SingleDeveloperView {

    private lateinit var binding: ActivityDetailBinding
    private lateinit var sharedInfo: GithubUsers

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Initialize ViewBinding
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val presenter = GithubPresenter()

        val githubUserName = intent.getStringExtra("gitUserName")
        val profileImage = intent.getStringExtra("profileImage")

        if (!githubUserName.isNullOrEmpty() && !profileImage.isNullOrEmpty()) {
            setProfile(githubUserName, profileImage)
            presenter.getDeveloperProfile(githubUserName, this)
        }

        binding.sharebutton.setOnClickListener {
            val shareIntent = Intent(Intent.ACTION_SEND).apply {
                type = "text/plain"
                putExtra(
                    Intent.EXTRA_TEXT,
                    "Checkout this awesome developer @${sharedInfo.userName}, ${sharedInfo.profile}."
                )
            }
            startActivity(Intent.createChooser(shareIntent, "Share Developer Info"))
        }
    }

    private fun setProfile(userName: String, profileImage: String) {
        binding.gitusername.text = userName
        Glide.with(this).asBitmap().load(profileImage).into(binding.image)
    }

    override fun showDeveloperProfile(profile: GithubUsers) {
        sharedInfo = profile
        binding.githuburl.text = profile.profile
        binding.org.text = profile.organization
    }
}